CMakeLists.txt

```diff
@@ -77,56 +86,54 @@ if(WIN32)
 endif()
 target_link_libraries(engine vk2d)
 target_link_libraries(engine asio)
-target_include_directories(engine PRIVATE 
-    "src/engine/" 
-)
+target_include_directories(engine PRIVATE ${INCLUDE_DIRS})
 
 # Create test executable
 add_executable(renderer_test src/test/renderer.test.cpp)
 target_link_libraries(renderer_test engine)
 target_include_directories(renderer_test PRIVATE 
-    "src/engine/"
+    ${INCLUDE_DIRS}
     "src/test/"
 )
 
 # Create test executable
 add_executable(gameObject_test src/test/gameObject.test.cpp)
 target_link_libraries(gameObject_test engine)
 target_include_directories(gameObject_test PRIVATE 
-    "src/engine/"
+    ${INCLUDE_DIRS}
     "src/test/"
 )
     
 # Test client
 add_executable(test_client src/test/testclient.cpp)
 target_link_libraries(test_client engine)
 target_include_directories(test_client PRIVATE 
+    ${INCLUDE_DIRS}
```

Comment:
Will be provided through the engine library if it includes the directory as `PUBLIC`.